### PR TITLE
Use array filter, update index numbers

### PR DESCRIPTION
**Pull request by acolombo1**
- Add a new JavaScript file and import it as a module related to the status updates.
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.
 (note: most of them were already fulfilled from the previous pull request)

### CODE REVIEW

File: src/modules/Tasklist.js [hunk 1]
```diff
@@ -178,36 +178,33 @@ export default class Tasklist {
...
-    if (changed) {
-      this.savedata();
-      this.#clearlist();
-      this.renderdata();
-    }
+    const result = this.tasks.filter((task) => task.completed === false);
+    this.tasks = [...result];
+    this.savedata();
+    this.#clearlist();
+    this.renderdata();
```
Comment (Cathella) [inline src/modules/Tasklist.js]: ✅ A function for the "Clear all completed" button using the `Array.filter()` method is implemented 👍🏻.

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @acolombo1 👋🏻, 

Your App is well styled, the interface looks beautiful 🤩. However, there are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct Github Flow.
✅ Professional README.
✅ Descriptive PR.
✅ It contains methods related to the status updates (completed: true / false).
✅ It has event listener on the checkbox (change).
✅ The task object's values for `completed` is updated upon user actions.

## Required Changes ♻️
_Check the comments under the review._

- [x] The array of items in local storage does not update after deleting an item. Kindly fix this issue.

image

- [x] The array index starts from 0, it is meant to start from 1. Please kindly fix this.

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by acolombo1** [comment]
Hope I understood what you meant, both issues have to do with the 'index' attribute of the object.
Fixed.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @acolombo1 👋🏻, 

Good job friend on making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**